src/sql/src/plan/query.rs

```diff
@@ -2488,16 +2488,33 @@ fn expand_select_item<'a>(
             Ok(items)
         }
         SelectItem::Wildcard => {
+            let mut pn = HashSet::new();
+            // Find all priority names
+            for item in ecx.scope.items.iter() {
+                if let Some(n) = item.names.get(0) {
+                    if n.priority {
+                        pn.insert(n.column_name.as_ref());
+                    }
+                }
+            }
+
             let items: Vec<_> = ecx
                 .scope
                 .items
                 .iter()
                 .enumerate()
-                .map(|(i, item)| {
-                    let name = item.names.get(0).and_then(|n| n.column_name.clone());
-                    (ExpandedSelectItem::InputOrdinal(i), name)
+                .filter_map(|(i, item)| {
+                    // If priority name and this is not a priority element, skip
+                    match item.names.get(0) {
+                        Some(n) if !n.priority && pn.contains(&n.column_name.as_ref()) => None,
+                        name => {
+                            let name = name.and_then(|n| n.column_name.clone());
+                            Some((ExpandedSelectItem::InputOrdinal(i), name))
+                        }
+                    }
                 })
                 .collect();
+
```

Comment:
Oh man, this scope stuff just gets messier and messier. This logic makes me fairly nervous. Can we instead add a `visible_to_wildcard: bool` to `ScopeItem` that's frobbed correctly by the outer join logic and just filter on that bool here?

Reply:
Yep, not a problem.